### DISCUSSION TITLE
Where can i add logging project?

### DISCUSSION THREAD

**Discussion by shailesh1984** [jasontaylordev/CleanArchitecture#802] [Q&A]
I need to add a logger in the clean architecture solution project, but I have a bit confused about where can I add a file/project in the solution.
Please help me with this.

Thank you

**Comment by tim1990s** [comment]
I suggest that the abstraction of the logging service put into the application layer.
The detail of implmentation shoul into the infrastructure.
Anyone else have another way?

**Reply by akshay-zz** [reply]
It all depends upon how many layers of decoupling an individual want in their application. If you want, then the above by @timnguyen90  is correct only.
